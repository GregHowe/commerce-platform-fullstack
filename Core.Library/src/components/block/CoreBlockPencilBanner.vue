<template>
	<div
		v-if="!bannerIsClosed"
		class="block__body"
		role="banner"
	>
		<div class="block__body-container">
			<div
				class="block__body-container_content"
				v-html="bodyCopy"
			></div>
			<CoreBlock
				v-if="label"
				class="block__body-container_button"
				:settings="{
					type: 'button',
					label,
					newTab: settings.newTab,
					url,
					icon: 'right-arrow',
					variants: {
						style: 'text',
						'icon-placement': 'icon-right',
					},
				}"
			/>
		</div>
		<CoreIcon
			v-if="settings.closeButton && !bannerIsClosed"
			icon="close"
			@click="bannerIsClosed = true"
		/>
	</div>
</template>

<script>
import { renderData } from "@libraryHelpers/dataComponents.js";

export default {
	name: "CoreBlockPencilBanner",
	props: {
		settings: {
			type: Object,
			required: true,
		},
		dataSite: {
			type: Object,
			default: () => ({}),
		},
	},
	data() {
		return {
			bannerIsClosed: false,
		};
	},
	computed: {
		bodyCopy() {
			return renderData(this.settings.bodyCopy, this.dataSite);
		},
		label() {
			return renderData(this.settings.buttonText, this.dataSite);
		},
		url() {
			return renderData(this.settings.buttonLink, this.dataSite);
		},
	},
};
</script>

<style lang="scss">
.block_pencil-banner {
	.block {
		width: unset;
	}
	.block__body-container {
		justify-content: inherit;
	}
	.block__body {
		display: flex;
		width: 100%;
		align-items: center;
		justify-content: space-between;
		padding: 0.3rem;
		background-color: var(--_core__pencil-banner_background);
		&-container {
			width: 100%;
			display: flex;
			align-items: center;
			flex-wrap: wrap;
			&_content {
				color: var(--_core__pencil-banner_color);
				font-family: var(--_core__pencil-banner_font-family);
				font-size: var(--_core__pencil-banner_font-size);
			}
			&_button {
				.block__body {
					margin: 0;
				}
			}
		}
	}
}
</style>
