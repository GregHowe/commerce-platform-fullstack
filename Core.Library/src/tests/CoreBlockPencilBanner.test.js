import CoreBlockPencilBanner from "../components/block/CoreBlockPencilBanner";
import "@testing-library/jest-dom";
import { render } from "@testing-library/vue";

const renderComponent = async (props = {}) => {
	return render(CoreBlockPencilBanner, {
		props: {
			...props,
			dataSite: {
				id: 666,
			},
		},
	});
};

describe("CoreBlockPencilBanner", () => {
	test("exists", async () => {
		await renderComponent({ settings: {} });
	});

	test("displays entered text", async () => {
		const { container } = await renderComponent({
			settings: { bodyCopy: "text across a banner" },
		});
		expect(container).toMatchSnapshot();
	});

	test("shows button if button text is entered", async () => {
		const { container } = await renderComponent({
			settings: { buttonText: "text in a button" },
		});
		expect(container).toMatchSnapshot();
	});

	test("shows close button if setting is true", async () => {
		const { container } = await renderComponent({
			settings: { closeButton: true },
		});
		expect(container).toMatchSnapshot();
	});
});
