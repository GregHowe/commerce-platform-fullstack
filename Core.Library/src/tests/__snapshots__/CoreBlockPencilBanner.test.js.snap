// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoreBlockPencilBanner displays entered text 1`] = `
<div>
  <div
    class="block__body"
    role="banner"
  >
    <div
      class="block__body-container"
    >
      <div
        class="block__body-container_content"
      >
        text across a banner
      </div>
       
      <!---->
    </div>
     
    <!---->
  </div>
</div>
`;

exports[`CoreBlockPencilBanner shows button if button text is entered 1`] = `
<div>
  <div
    class="block__body"
    role="banner"
  >
    <div
      class="block__body-container"
    >
      <div
        class="block__body-container_content"
      />
       
      <div
        class="block__body-container_button block block_button block_button_text block_button_icon-right"
        target="_blank"
      >
        <div
          class="block__body"
          index="0"
          nested-level="0"
        >
          <a
            class="button"
            role="button"
          >
            <span
              class="button_icon--custom"
            >
              <span
                class="button_icon"
                icon="right-arrow"
                name="buttonIcon"
              />
            </span>
             
            <span
              class="button_label button_icon--down"
            >
              <span
                class="button_label_text"
              >
                text in a button
              </span>
               
              <span
                class="button_icon button_i"
                icon="down-arrow"
              />
            </span>
          </a>
        </div>
          
        <!---->
      </div>
    </div>
     
    <!---->
  </div>
</div>
`;

exports[`CoreBlockPencilBanner shows close button if setting is true 1`] = `
<div>
  <div
    class="block__body"
    role="banner"
  >
    <div
      class="block__body-container"
    >
      <div
        class="block__body-container_content"
      />
       
      <!---->
    </div>
     
    <span
      icon="close"
    />
  </div>
</div>
`;
